### ISSUE TITLE
Tectonic Bare Metal mode fails if your disk is not named /dev/sda

### ISSUE THREAD

**Issue by arianvp** [coreos/tectonic-forum#142]
### Tectonic Version 
`tectonic-1.6.2-tectonic-1`

### Environment

https://vultr.com  VPS servers and Dedicated servers over `iPXE` (Bare Metal)

### Expected Behavior
`installer.service`  runs `coreos-install` which partitions the disk and then reboots the machine

### Actual Behavior
`coreos-install` fails because it cannot find the disk `/dev/sda`

### Reproduction Steps

1. Have a datacenter where the disk that your sysadmins give you are not labelled /dev/sda
2. Use the Tectonic UI
3. Select "Bare Metal" mode. Install matchbox on your DHCP server
4. Let Tectonic provision matchbox with terraform
5.  IPXE boot some machines in your data center
6. See them fail one by one because `installer.service` fails

### Workaround:

1. Press the scary "Advanced mode: Manually boot your own cluster. Validate configuration, generate assets, but don't create the cluster."

2. Unzip the "assets.zip" that the UI provides you

3.  Edit " ./platforms/metal/cl/coreos-install.yaml.tmpl"
    and replace `/dev/sda` by `/dev/vda` or whatever name you need

4. 
  ```
  terraform get ./platforms/metal
  terraform plan -var-file=./terraform.tfvars platforms/metal
  terraform apply -var-file=./terraform.tfvars platforms/metal
  ```

5. Wake up all your machines and let them boot over IPXE
6. CoreOS Installs nicely

### Desired Feature

Make the disk to which `coreos-install` applies its actions configurable in **Tectonic UI**.

Either *per node*  or *per cluster*
For me it would be enough to just set this cluster-wide.  All our disk are called `/dev/vda`.
But I can imagine that people would like even more

**Comment by arianvp** [comment]
Have not actually tried the workaround yet because it's sleepy time. I will try it first thing in the morning

**Comment by arianvp** [comment]
Okay I tried it but it did not work.

Installation of the `master` nodes is failing with the following error:

```
Jun 07 21:28:17 node1.**** bash[4532]: [  954.779097] hyperkube[5]: TPR prometheus.monitoring.coreos.com not available yet, retrying in 5 seconds
Jun 07 21:28:23 node1.**** bash[4532]: [  960.601266] hyperkube[5]: Error from server (NotFound): thirdpartyresources.extensions "prometheus.monitoring.coreos.com" not found
Jun 07 21:28:23 node1.**** bash[4532]: [  960.616811] hyperkube[5]: TPR prometheus.monitoring.coreos.com not available yet, retrying in 5 seconds

```

**Comment by arianvp** [comment]
The other issue is totally unrelevant. Ignore it. It has to do with kubernetes not being able to find a place to schedule prometheus stuff due to insufficient resources. :)

**Comment by robszumski** [comment]
@arianvp just to be clear, modifying the install files worked? If so, that's the benefit of having this installer open source and using Terraform :)

As none of our officially supported platforms requires this change, I don't think there is anything actionable right now. We're always open to adding more platforms as demand dictates and as we have more automation bandwidth to properly support them going forward.

**Comment by arianvp** [comment]
Yes, editing `./platforms/metal/cl/coreos-install.yaml.tmpl` and replacing `/dev/sda/` with `/dev/vda/` got everything working :tada: 

But the 'bare metal' mode is a 'supported platform' right? If so, I think it is a small change. However, perhaps more things need to be configured for people's custom bare metal setups except for  the disk name, so this might be a case of "once we add this ,we need to add stuff for other custom configurations as well, which we don't feel like maintaining" which is a sensible sentiment. 

However, if you decide not to add this to the Bare Metal UI flow, it would be nice to add some disclaimer that the Bare Metal flow might require some careful customization to work with your on premise setup.  Or advice people to go the terraform road directly instead.

**Comment by abh** [comment]
See also #15
